web.rb

```diff
@@ -33,19 +33,24 @@
 
 post '/charge' do
   authenticate!
-  # Get the credit card details submitted by the form
-  source = params[:source]
-  customer = params[:customer_id] || @customer.id
-  
+  # Get the credit card details submitted
+  payload = params
+  if request.accept? 'application/json' and params.empty? 
```

Comment:
`accept?` isn't what you want, unfortunately.

`if request.content_type == 'application/json'` would be better.